pkg/statsd/statsd.go

```diff
@@ -46,14 +49,32 @@ type Server struct {
 
 // Run runs the server until context signals done.
 func (s *Server) Run(ctx context.Context) error {
-	return s.RunWithCustomSocket(ctx, func() (net.PacketConn, error) {
-		return net.ListenPacket("udp", s.MetricsAddr)
-	})
+	return s.RunWithCustomSocket(ctx, socketFactory(s.MetricsAddr, s.ConnPerReader))
 }
 
 // SocketFactory is an indirection layer over net.ListenPacket() to allow for different implementations.
 type SocketFactory func() (net.PacketConn, error)
 
+func socketFactory(metricsAddr string, connPerReader bool) SocketFactory {
+	if connPerReader {
+		// go-reuseport requires explicitly representing the unspecified address
+		addr, err := net.ResolveUDPAddr("udp", metricsAddr)
```

Comment:
It was either this, or a breaking change to flags to `metrics-addr` to require specifying the IP